community/cypher/cypher-compiler-3.1/src/test/scala/org/neo4j/cypher/internal/compiler/v3_1/planner/logical/OptionalMatchRemoverTest.scala

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (c) 2002-2016 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.cypher.internal.compiler.v3_1.planner.logical
+
+import org.neo4j.cypher.internal.compiler.v3_1.SyntaxExceptionCreator
+import org.neo4j.cypher.internal.compiler.v3_1.ast.convert.plannerQuery.StatementConverters.toUnionQuery
+import org.neo4j.cypher.internal.compiler.v3_1.ast.rewriters.flattenBooleanOperators
+import org.neo4j.cypher.internal.compiler.v3_1.planner._
+import org.neo4j.cypher.internal.frontend.v3_1.Rewritable._
+import org.neo4j.cypher.internal.frontend.v3_1.ast.Query
+import org.neo4j.cypher.internal.frontend.v3_1.helpers.fixedPoint
+import org.neo4j.cypher.internal.frontend.v3_1.test_helpers.CypherFunSuite
+import org.neo4j.cypher.internal.frontend.v3_1.{DummyPosition, SemanticTable}
+
+class OptionalMatchRemoverTest extends CypherFunSuite with LogicalPlanningTestSupport2 {
```

Comment:
This is lovely; would be nice to add some complex examples with disconnected patterns.